### ISSUE TITLE
When leveraging IAspect with Python, visit() method doesn't pass an IConstruct but an ObjRef

### ISSUE THREAD

**Issue by fitzee** [aws/aws-cdk#3630]
**Note: for support questions, please first reference our [documentation](https://docs.aws.amazon.com/cdk/api/latest), then use [Stackoverflow](https://stackoverflow.com/questions/ask?tags=aws-cdk)**. This repository's issues are intended for feature requests and bug reports.

* **I'm submitting a ...**
  - [x] :beetle: bug report
  - [ ] :rocket: feature request
  - [ ] :books: construct library gap
  - [ ] :phone: security issue or vulnerability => Please see [policy](https://github.com/aws/aws-cdk/security/policy)
  - [ ] :question: support request => Please see note at the top of this template.

* **What is the current behavior?**
*If the current behavior is a :beetle:bug:beetle:: Please provide the steps to reproduce*

``` py
@jsii.implements(IAspect)
class VisitorTest:
    def visit(self, node: "IConstruct") -> None:
        print(type(node))
```

node is an ObjRef not an IConstruct, as such I can't do something like:

`node.node.path `

within my visit() method

* **What is the expected behavior (or behavior of feature suggested)?**
I expect visit() on a class that implements IAspect to actually be passed an instance of a IConstruct, but I am getting an ObjRef instead.  There is no node() property on the argument currently being passed.

* **What is the motivation / use case for changing the behavior or adding this feature?**

* **Please tell us about your environment:**

  - CDK CLI Version: 1.3.0
  - Module Version: 1.3.0
  - OS: OSX Mojave
  - Language: Python (3.7.3)

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. associated pull-request, stackoverflow, gitter, etc)

**Comment by RomainMuller** [comment]
Yes - this is a bug we've just stumbled on the root cause by accident. The next CDK release should have this fixed!

**Comment by RomainMuller** [comment]
This has been fixed now (CDK `>= 1.17.0`). Resolving.
